src/Hashing/Argon2IdHashing.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+/**
+ * This file is part of the Zest Framework.
+ *
+ * @author   Muhammad Umer Farooq <lablnet01@gmail.com>
+ * @author-profile https://www.facebook.com/Muhammadumerfarooq01/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ * @since 3.0.0
+ *
+ * @license MIT
+ */
+
+namespace Zest\Hashing;
+
+class Argon2IdHashing extends AbstractHashing
+{
+    /**
+     * The default memory cost.
+     *
+     * @since 3.0.0
+     *
+     * @var int
+     */
+    private $memory = 1024;
+
+    /**
+     * The default time cost.
+     *
+     * @since 3.0.0
+     *
+     * @var int
+     */
+    private $time = 2;
+
+    /**
+     * The default threads cost.
+     *
+     * @since 3.0.0
+     *
+     * @var int
+     */
+    private $threads = 2;
+
+    /**
+     * Determine whether to perform an algorithm check.
+     *
+     * @since 3.0.0
+     *
+     * @var bool
+     */
+    private $verifyAlgorithm = false;
+
+    /**
+     * __construct.
+     *
+     * @param (array) $options
+     *
+     * @since 3.0.0
+     */
+    public function __construct(array $options)
+    {
+        $this->setMemory($options['memory'])->setTime($options['time'])->setThreads($options['threads']);
```

Comment:
I think it needs to check whether the keys are existed in specific `$options` array.

Consider following code snippets:

```php
$this->setMemory($options['memory'] ?? $this->memory)->setTime($options['time'] ?? $this->time)->setThreads($options['threads'] ?? $this->threads);
```

Should we need the `$options` validation?

Reply:
Yes, better we can check, but i think these are required

Reply:
Yes , so i think we can  keep simple our setMemory() etc functions

Reply:
If we keep the related `set*` methods simple, we also have to consider the `BcryptHashing::setCost` method and its constructor.

Reply:
i have push commit to master branch for `BcryptHashing`